src/main/java/randoop/condition/ConditionMethodCreator.java

```diff
@@ -0,0 +1,204 @@
+      if (diag != null) {
+        String diagMessage = diag.getMessage(null);
+        if (diagMessage.contains("unreported exception")) {
+          msg.append(
+              String.format(
+                  "condition threw exception %s",
+                  diagMessage.substring(0, diagMessage.indexOf(';'))));
+        } else {
+          msg.append(diagMessage);
+        }
+      }
+    }
+    msg.append(String.format("%nClass Declaration:%n%s", classText));
+    return msg.toString();
+  }
+
+  /**
+   * Create the source code for the condition class.
+   *
+   * @param methodName the name of the condition method
+   * @param conditionText the condition source code
+   * @param parameterDeclarations the signature string for the condition method
+   * @param packageName the package of the condition class
+   * @param conditionClassName the name of the condition class
+   * @return the Java source code for the condition class
+   */
+  private static String createConditionClassSource(
+      String methodName,
+      String conditionText,
+      String parameterDeclarations,
+      String packageName,
+      String conditionClassName) {
+    String classSource = "";
+    if (packageName != null && !packageName.isEmpty()) {
+      classSource = "package " + packageName + ";" + Globals.lineSep;
+    }
+    return classSource
+        + Globals.lineSep
+        + "public class "
+        + conditionClassName
+        + " {"
+        + Globals.lineSep
+        + "  public static boolean "
+        + methodName
+        + parameterDeclarations
+        + " throws Throwable {"
+        + Globals.lineSep
+        + "    return "
+        + conditionText
+        + ";"
+        + Globals.lineSep
+        + "  }"
+        + Globals.lineSep
+        + "}"
+        + Globals.lineSep;
+  }
+
+  /**
+   * Creates the {@link RawSignature} for the precondition method.
+   *
+   * <p>if {@code shift == 1}, the parameter types for the condition method have the receiver type
+   * first, followed by the parameter types. Otherwise, the condition method parameter types are
+   * just the parameter types.
+   *
+   * <p>Note that these signatures may be used more than once for different condition methods, and
+   * so {@link #create(RawSignature, String, String, SequenceCompiler)} replaces the classname to
+   * ensure a unique name.
+   *
+   * @param packageName the package name for the condition class
+   * @param receiverType the declaring class of the method or constructor, used as receiver type if
+   *     {@code shift == 1}
+   * @param parameterTypes the parameter types for the original method
+   * @param shift starting position of {@code parameterTypes} in condition method parameter list,
+   *     either 0 or 1
+   * @return the constructed pre-condition method signature
+   */
+  static RawSignature getPreconditionSignature(
+      String packageName, Class<?> receiverType, Class<?>[] parameterTypes, int shift) {
```

Comment:
I think it would be easier to understand the method specification (and less internal checking would be necessary) if this were changed to a boolean.